android-components/components/feature/search/src/main/java/mozilla/components/feature/search/telemetry/SearchProviderModel.kt

```diff
@@ -7,45 +7,60 @@ package mozilla.components.feature.search.telemetry
 /**
  * All data needed to identify ads of a particular provider.
  *
- * @property name provider name.
- * @property regexp regular expression used to match the provider.
- * @property queryParam name of the query parameter for the user's search string.
- * @property codeParam name of the query parameter for the partner code.
- * @property codePrefixes array of prefixes (or complete values) to match against
+ * @property telemetryId provider name.
+ * @property searchPageRegexp regular expression used to match the provider.
+ * @property expectedOrganicCodes array of partner codes to match against the parameters in the url.
+ * Matching these codes will report the SERP as organic:none which means the user has done a search
+ * through the search engine's website rather than through SAP.
+ * @property queryParamName name of the query parameter for the user's search string.
+ * @property codeParamName name of the query parameter for the partner code.
+ * @property taggedCodes array of prefixes (or complete values) to match against
  * the partner code parameters in the url.
- * @property followOnParams array of query parameter names that are used when a follow-on search occurs.
+ * @property followOnParamNames array of query parameter names that are used when a follow-on search occurs.
  * @property extraAdServersRegexps array of regular expressions that match URLs of potential ad servers.
  * @property followOnCookies array of cookie details that are used to identify follow-on searches.
  */
 internal data class SearchProviderModel(
-    val name: String,
-    val regexp: Regex,
-    val queryParam: String,
-    val codeParam: String,
-    val codePrefixes: List<String>,
-    val followOnParams: List<String>,
+    val schema: Long,
```

Comment:
nit: not necessary, but it would be nice if these were ordered in the same order as the json